packages/itwinui-react/src/core/Table/Table.tsx

```diff
@@ -822,7 +826,7 @@ export const Table = <
 
   useLayoutEffect(() => {
     if (scrollToIndex) {
-      virtualizer.scrollToIndex(scrollToIndex, { align: 'center' });
+      virtualizer.scrollToIndex(scrollToIndex, { align: 'start' });
```

Comment:
Correct me if I'm wrong, but this is what happens when using virutalization:
1. `scrollIntoView` (from `useScrollToRow`) executes first, but it fails because the row does not exist in the DOM.
2. `virtualizer.scrollToIndex` is what actually scrolls the row to the top, but it still stays hidden underneath the Table header.
3. `scrollTo` (within `setTimeout` in `useScrollToRow`) executes and finally shows the Table.

Feels a bit indirect, no? See tanstack-virtual's [`scrollMargin`](https://tanstack.com/virtual/latest/docs/api/virtualizer#scrollmargin)  option for a more direct solution.